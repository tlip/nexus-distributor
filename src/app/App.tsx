import React, { useEffect } from 'react';
import { ThemeProvider } from 'theme-ui';
import { Flex } from 'components/Flex';
import { Card } from 'components/Card';
import { Text } from 'components/Text';
import { Button } from 'components/Button';
import { theme } from 'theme';
import { useAsyncRates } from 'state/hooks';

export const App: React.FC = () => {
  const [rates, fetchRates] = useAsyncRates();

  useEffect(() => {
    fetchRates();
  }, []);
  console.log({ rates });

  return (
    <ThemeProvider theme={theme}>
      <Flex
        justifyContent="center"
        alignItems="center"
        width="100vw"
        height="100vh"
        bg="background"
      >
        <Card width="100%" maxWidth="20em">
          <Text variant="h4">Hello, welcome to Nexus Mutual</Text>
          <Text as="p" mt="2em">
            I think i'm just text. That's pretty cool, right?
          </Text>
          <Text as="div" mt="1em" mb="2em">
            Test my buttons!
          </Text>
          <Button width="100%" my="0.5em">
            Primary Button
          </Button>
          <Button variant="outline" width="100%">
            Outline Button
          </Button>
        </Card>
      </Flex>
    </ThemeProvider>
  );
};
