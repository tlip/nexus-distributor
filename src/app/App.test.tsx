import React from 'react';
import { render } from '@testing-library/react';
import { App } from './App';

describe('App component', () => {
  it('should render without crashing', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
