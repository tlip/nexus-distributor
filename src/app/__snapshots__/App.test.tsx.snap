// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render without crashing 1`] = `
<div>
  <div
    class="css-b3psm6-Flex e11m02v90"
  >
    <div
      class="css-cu2094-Box e1gx7u5e0"
    >
      <h4
        class="css-1cfxgj"
      >
        Hello, welcome to Nexus Mutual
      </h4>
      <p
        class="css-gn5fqb"
      >
        I think i'm just text. That's pretty cool, right?
      </p>
      <div
        class="css-3q7i21"
      >
        Test my buttons!
      </div>
      <button
        class="css-1ivwkj0"
      >
        <span
          class="e7emh9v1 css-cj61wh-ButtonText"
        >
          Primary Button
        </span>
      </button>
      <button
        class="css-1v411tv"
      >
        <span
          class="e7emh9v1 css-cj61wh-ButtonText"
        >
          Outline Button
        </span>
      </button>
    </div>
  </div>
</div>
`;
